### PR TITLE
[UILISTS-122] Provide list of current columns to query builder when duplicating

### PR DESCRIPTION
**Pull request by ncovercash**
The majority of this PR is cleaning up some TypeScript issues while poking around here. Mainly, it was to create some interfaces for requests to create/update lists, making it nicer/type-safe to send update/create requests

### CODE REVIEW

File: src/pages/copylist/CopyListPage.tsx [hunk 1]
```diff
@@ -121,6 +120,7 @@ export const CopyListPage:FC = () => {
             status={state[FIELD_NAMES.STATUS]}
             visibility={state[FIELD_NAMES.VISIBILITY]}
             description={state[FIELD_NAMES.DESCRIPTION]}
+            recordColumns={listDetails?.fields}
```
Comment (ncovercash) [inline src/pages/copylist/CopyListPage.tsx:123]: In truth, this is the only "real" change in this PR.

File: src/components/ConfigureQuery/ConfigureQuery.tsx [hunk 2]
```diff
@@ -112,16 +122,12 @@ export const ConfigureQuery:FC<ConfigureQueryProps> = (
     return ky.delete(`query/${queryId}`);
   };
 
-  const onColumnChange = (columns: string[]) => {
```
Comment (ncovercash) [inline src/components/ConfigureQuery/ConfigureQuery.tsx:115]: This callback was redundant

File: src/components/ConfigureQuery/ConfigureQuery.tsx [hunk 3]
```diff
@@ -59,31 +59,41 @@ export const ConfigureQuery:FC<ConfigureQueryProps> = (
     return ky.get(`query/${queryId}`, { searchParams }).json();
   };
 
-  const testQueryDataSource = async ({ fqlQuery } : { fqlQuery: any }) => {
+  const testQueryDataSource = async ({ fqlQuery } : { fqlQuery: FqlQuery }) => {
```
Comment (ncovercash) [inline src/components/ConfigureQuery/ConfigureQuery.tsx:62]: at least with these, now TS can tell between object-form and stringified

File: src/components/ConfigureQuery/ConfigureQuery.tsx [hunk 4]
```diff
@@ -48,80 +54,111 @@ export const ConfigureQuery:FC<ConfigureQueryProps> = (
...
+    entityTypeId,
+    columnName,
+    searchValue,
+  }: {
+    entityTypeId: string;
+    columnName: string;
+    searchValue: string;
+  }) => {
+    return ky
+      .get(`entity-types/${entityTypeId}/columns/${columnName}/values?search=${searchValue}`)
```
Comment (ncovercash) [inline src/components/ConfigureQuery/ConfigureQuery.tsx:148]: this was another bug i saw here - if `searchValue` contains something like `foo?bar`, it'll only actually get `foo`. i'm going to spin this off into a separate ticket though

**Review by mweaver-ebsco — APPROVED** [review]

File: src/components/ConfigureQuery/ConfigureQuery.tsx (hunk 4 above)
Comment (ncovercash) [reply src/components/ConfigureQuery/ConfigureQuery.tsx:143]: spun off to https://folio-org.atlassian.net/browse/UILISTS-124

**Review by bvsharp — APPROVED** [review]

**Pull request merged**